x-pack/plugins/code/server/indexer/index_version_controller.ts

```diff
@@ -9,29 +9,45 @@ import _ from 'lodash';
 import { IndexMigrator } from '.';
 import { EsClient } from '../lib/esqueue';
 import { Logger } from '../log';
+import { EsException } from './es_exception';
 import { IndexCreationRequest } from './index_creation_request';
 import pkg from './schema/version.json';
 
 export class IndexVersionController {
   private version: number;
+  private DEFAULT_VERSION = 0;
 
   constructor(protected readonly client: EsClient, private readonly log: Logger) {
     this.version = Number(pkg.codeIndexVersion);
   }
 
   public async tryUpgrade(request: IndexCreationRequest) {
     this.log.debug(`Try upgrade index mapping/settings for index ${request.index}.`);
-    const esIndexVersion = await this.getIndexVersionFromES(request.index);
-    const needUpgrade = this.needUpgrade(esIndexVersion);
-    if (needUpgrade) {
-      const migrator = new IndexMigrator(this.client, this.log);
-      const oldIndexName = `${request.index}-${esIndexVersion}`;
-      this.log.warn(
-        `Migrate index mapping/settings from version ${esIndexVersion} for ${request.index}`
-      );
-      return migrator.migrateIndex(oldIndexName, request);
-    } else {
-      this.log.debug(`Index version is update-to-date for ${request.index}`);
+    try {
+      const esIndexVersion = await this.getIndexVersionFromES(request.index);
+      const needUpgrade = this.needUpgrade(esIndexVersion);
+      if (needUpgrade) {
+        const migrator = new IndexMigrator(this.client, this.log);
+        const oldIndexName = `${request.index}-${esIndexVersion}`;
+        this.log.info(
+          `Migrate index mapping/settings from version ${esIndexVersion} for ${request.index}`
+        );
+        return migrator.migrateIndex(oldIndexName, request);
+      } else {
+        this.log.debug(`Index version is update-to-date for ${request.index}`);
+      }
+    } catch (error) {
+      if (
+        error &&
```

Comment:
why this branch you check `error` but in the previous branch you didn't?

Reply:
`error` check here is not quite necessary.